src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -628,7 +628,8 @@ AdaptiveExecutor(CitusScanState *scanState)
 	DistributedPlan *distributedPlan = scanState->distributedPlan;
 	EState *executorState = ScanStateGetExecutorState(scanState);
 	ParamListInfo paramListInfo = executorState->es_param_list_info;
-	TupleDesc tupleDescriptor = ScanStateGetTupleDescriptor(scanState);
+	CustomScan *remoteScan = ((CustomScan *)scanState->customScanState.ss.ps.plan);
+	TupleDesc tupleDescriptor = ExecCleanTypeFromTL(remoteScan->custom_scan_tlist);//ScanStateGetTupleDescriptor(scanState);
```

Comment:
In pg11 this has a 2nd parameter `bool hasoid`

Reply:
Thanks! I have been prototyping this on pg12. Have a sync with Andres tomorrow about this approach, hence the reason I pushed this without pg11 working.

After we make sure this is a sane approach to move forward with I will take this into account during the refactoring.